app/controllers/multifactor_auths_controller.rb

```diff
@@ -1,8 +1,13 @@
 class MultifactorAuthsController < ApplicationController
+  include MfaExpiryMethods
+  include WebauthnVerifiable
+
   before_action :redirect_to_signin, unless: :signed_in?
-  before_action :require_mfa_disabled, only: %i[new create]
-  before_action :require_mfa_enabled, only: :update
+  before_action :require_totp_disabled, only: %i[new create]
+  before_action :require_mfa_enabled, only: %i[update destroy]
   before_action :seed_and_expire, only: :create
+  before_action :verify_session_expiration, only: %i[mfa_update webauthn_update]
+  after_action :delete_mfa_level_update_session_variables, only: %i[mfa_update webauthn_update]
```

Comment:
Added a before action to verify session expiration and edited the after action to delete all relevant session variables.